salt/netapi/rest_cherrypy/app.py

```diff
@@ -1210,29 +1210,21 @@ def GET(self, jid=None, timeout=''):
         '''
         lowstate = [{
             'client': 'runner',
-            'fun': 'jobs.lookup_jid' if jid else 'jobs.list_jobs',
+            'fun': 'jobs.list_job' if jid else 'jobs.list_jobs',
             'jid': jid,
         }]
 
-        if jid:
-            lowstate.append({
-                'client': 'runner',
-                'fun': 'jobs.list_job',
-                'jid': jid,
-            })
-
         cherrypy.request.lowstate = lowstate
         job_ret_info = list(self.exec_lowstate(
             token=cherrypy.session.get('token')))
 
         ret = {}
         if jid:
-            job_ret, job_info = job_ret_info
-            ret['info'] = [job_info]
+            ret['info'] = job_ret_info[0]
+            ret['return'] = [dict((k, job_ret_info[0]['Result'][k]['return']) for k in job_ret_info[0]['Result'])]
```

Comment:
The intent is to preserve the original format for clients of the API that
might depend on it. The comprehension converts the contents of the Results
key from a lookup_job query to be consistent with the results from
lookup_jid; so from:

Result:
    ----------
    host1:
        ----------
        return:
            True
    host2:
        ----------
        return:
            True

to:

host1:
    True
host2:
    True





On Mon, Aug 22, 2016 at 11:05 AM, Seth House <notifications@github.com>
wrote:

> In salt/netapi/rest_cherrypy/app.py
> <https://github.com/saltstack/salt/pull/35665#discussion_r75727517>:
>
> >          cherrypy.request.lowstate = lowstate
> >          job_ret_info = list(self.exec_lowstate(
> >              token=cherrypy.session.get('token')))
> >
> >          ret = {}
> >          if jid:
> > -            job_ret, job_info = job_ret_info
> > -            ret['info'] = [job_info]
> > +            ret['info'] = job_ret_info[0]
> > +            ret['return'] = [dict((k, job_ret_info[0]['Result'][k]['return']) for k in job_ret_info[0]['Result'])]
>
> I'm not sure I follow the intent of this list comprehension. Would the
> following suffice?
>
> job_info = job_ret_info[0]
> job_ret = job_info.get('Result', {})
> [...]
> ret['return'] = [job_ret]
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/saltstack/salt/pull/35665/files/25b43befaa5183b3e6da07066732cdc8e10689dc#r75727517>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHKLyTdGe6bALils2XTakSnUsqG2x3v0ks5qieT-gaJpZM4JqHAh>
> .
>